### PR TITLE
OS-8326 fix compilation with GCC10

### PR DESCRIPTION
**Pull request by danmcd**
Part of OS-8326...

### CODE REVIEW

**Comment by danmcd** [comment]
Will need testing directions.  As far as I can tell, all I'm doing is providing hints that gcc10 does 64-bit by default, and we should be more explicit in our guidance to ld(1).  See illumos#14129 for an analogous fix.

**Review by bahamat — DISMISSED** [review]

**Comment by danmcd** [comment]
At this point, I can push this independently of the illumos-extra push, I believe.  Please re-confirm this @bahamat ?

File: CHANGES.md [hunk 1]
```diff
@@ -10,6 +10,10 @@
 
 # mdb_v8 changelog
 
+## v1.4.3 (2021-12)
+
+* #xxx fix compilation with GCC10
```
Comment (bahamat) [inline CHANGES.md:15]: This needs the actual ticket number

**Comment by bahamat** [comment]
Yeah, this should be fine to go, but we need `CHANGES.md` updated accurately.

**Comment by danmcd** [comment]
Should be fixed now (and rebased, and with the OS-8326 summary in the commit message).

**Review by bahamat — APPROVED** [review]
OK, this LGTM.

**Pull request merged**